### DISCUSSION TITLE
How to get relative paths in the generated site

### DISCUSSION THREAD

**Discussion by MGathier** [nuxt/nuxt#25377] [Questions]
We are developing an application with Nuxt. This application can be deployed behind a proxy with a URL mapping, for instance the proxy maps requests starting with /app1/ to our application. 
The application does not use server-side rendering (ssr is false in Nuxt config). 

When I generate the application, the index.html file has absolute paths for the javascript and stylesheets. 
Then I tried to set the baseURL to "." and buildAssetsDir to "assets", which seems to output the correct code, but when I run this code with /app1/ mapped, I got "This page you are looking for could not be found.".

Am I missing something here?

**Comment by s0h311** [comment]
What about?:

```TypeScript
export default defineNuxtConfig({
  ...
  nitro: {
    baseURL: '/app1/'
  }
  ...
})
```

https://nitro.unjs.io/config#baseurl

**Comment by MGathier** [comment]
That would work if I would have control over the URL, but as this is part of a standard product it is up to the customer to define the base URL. The specific usecase is where the application has a number of instances running in Docker/K8s and the user wants to connect to a specific instance. In this case the proxy would have a different URL per instance.

**Reply by s0h311** [reply]
What about a wildcard? Does `/*/` work?

**Comment by xjccc** [comment]
like this？
```
app: {
  baseURL: '/app1/'
}
```

**Comment by DMiradakis** [comment]
Came across this question today as I'm building an ElectronJS app that uses a Nuxt app inside. I use the Nuxt dev server during development, but for production & packaging I need to build the Nuxt app with `nuxt generate` and bundle the build assets into the Electron app for distribution.

I was having the same issue where the asset filepaths to the `_nuxt` assets folder were breaking with Electron's `file://` protocol that it uses to serve the built index.html file.

I found this [Stack Overflow question](https://stackoverflow.com/questions/69657203/how-can-i-make-paths-in-generated-nuxt-app-index-html-relative) that suggested setting the `app.cdnURL` property in `nuxt.config.ts` to `./` like below:

```json
app: { cdnURL: './' }
````

I can confirm this works for Nuxt v4.3.0 in my Electron app.
